# 6B. ATAC and PRO-seq histograms
The following code was used to make histograms from ATAC-seq and PRO-seq data in bash and R.

## ATAC-seq histograms
ATAC-seq histograms were made using DeepTools and R.
### BamCoverage
BAM files were normalizeed using the reciprocal sizefactor from DEseq2 in Deeptools in bash.
```
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0002.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0002.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 1.045911761 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0003.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0003.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 0.967167282 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0004.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0004.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 1.04100281 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0006.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0006.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 0.810278939 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0007.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0007.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 1.063125749 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0009.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0009.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 0.97279605 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0010.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0010.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --binSize 10 --scaleFactor 0.933970874 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0011.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0011.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --numberOfProcessors 10 --binSize 10 --scaleFactor 1.078170719 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
bamCoverage --bam /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0012.hg19.sorted.F4q10.bam -o /Volumes/Hillary_2p/9601_ATAC_DHL4/9601-MB-0012.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --numberOfProcessors 10 --binSize 10 --scaleFactor 1.087157528 --effectiveGenomeSize 2827437033 --blackListFileName /Volumes/Hillary_2p/genomes/hg19_blacklist/hg19.blacklistpeaks.bed --extendReads
```
### Calculate signal around peaks
The signal around the desired peak set was calculated and plotted using Deeptools in bash.
```
computeMatrix reference-point --referencePoint center -b 3000 -a 3000 -R /Volumes/Hillary_X6/Nud_DHL4_ATAC_forTobias/TF_footprinting_out/20230801_funcFKHmotifs_motifonly.bed -S 6438/6438-HL-1_0hrA.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-2_0hrB.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-5_15minA.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-6_15minB.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-7_30minA.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-8_30minB.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-9_1hrA.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/6438-HL-10_1hrB.hg19dm3.F4q10.sorted.dm3rm.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/9330-HL-0011_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 6438/9330-HL-0012_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0001_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0002_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0003_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0004_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0005_S1_L005_samp81mill.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0006_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0007_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0008_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0009_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9330-NUDUL1_ATAC/9330-HL-0010_S1_L005.hg19dm3.F4q10.sorted.chrMrm.sorted.sfdeseqrecip.bs10.bw 9601_ATAC_DHL4/9601-MB-0002.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0003.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0004.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0006.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0007.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0009.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0010.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0011.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw 9601_ATAC_DHL4/9601-MB-0012.hg19.sorted.F4q10.sfdeseqrecip_2.bs10.bw --skipZeros --binSize 10 -o 20230826_ATAC_LY1_NUD_DHL4_computematrix_funcFKHmotifs_ba3000_bs10.gz

plotProfile --matrixFile 20230826_ATAC_LY1_NUD_DHL4_computematrix_funcFKHmotifs_ba3000_bs10.gz --outFileName 20230826_ATACsig_functionalFKH_bs10.svg --plotHeight 15 --plotWidth 15 --plotType lines --perGroup --yAxisLabel "ATAC Signal" --regionsLabel "Functional FKH Motifs" --refPointLabel "Motif Center" --outFileNameData 20230826_ATAC_LY1_NUD_DHL4_computematrix_funcFKHmotifs_ba3000_bs10_profile.txt
```
### Plot histograms
Final histograms were plotted in R.
#### Load libraries
```{r}
library(tidyverse)
library(hillaryscolors)
library(viridis)
```
#### Load Files
```{r}
all <- read.delim("20230826_ATAC_LY1_NUD_DHL4_computematrix_funcFKHmotifs_ba3000_bs10_profile_adj.txt") %>%
  select(1:602) %>%
  pivot_longer(X1:X600, names_to = "bin", values_to = "score") %>%
  filter(Sample != "LY1_0.5_B") %>%
  mutate(bin = parse_number(bin)) %>%
  pivot_wider(names_from = Sample, values_from = score)

all_peak <- read.delim("20230826_ATAC_LY1_NUD_DHL4_computematrix_funcFKHpeaks_ba3000_bs10_profile_adj.txt") %>%
  select(1:602) %>%
  pivot_longer(X1:X600, names_to = "bin", values_to = "score") %>%
  filter(Sample != "LY1_0.5_B") %>%
  mutate(bin = parse_number(bin)) %>%
  pivot_wider(names_from = Sample, values_from = score)
```

#### Make histograms
```{r}
## load histpalette
viridis::inferno(6)

histpalette <- c("0 hr" = "#5F5F60", "15 min" = "#420A68FF", "30 min" = "#932667FF", "1 hr" = "#DD513AFF", "2 hr" = "#FCA50AFF")

NUD_hist <- ggplot (all, aes(x = bin)) +
  geom_line(aes(y = NUD_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = NUD_0.25_A, color = "15 min"), size = 0.75) +
  geom_line(aes(y = NUD_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = NUD_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = NUD_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 600, by = 200), expand = c(0, 0), limits = c(0, 600)) +
  ggtitle("NUDUL1 PS Around Func Motif") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
NUD_hist

LY1_hist <- ggplot (all, aes(x = bin)) +
  geom_line(aes(y = LY1_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = LY1_0.25_A, color = "15 min"), size = 0.75) +
  geom_line(aes(y = LY1_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = LY1_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = LY1_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 600, by = 200), expand = c(0, 0), limits = c(0, 600)) +
  ggtitle("LY1 PS Around Func Motif") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
LY1_hist

DHL4_hist <- ggplot (all, aes(x = bin)) +
  geom_line(aes(y = DHL4_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = DHL4_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = DHL4_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = DHL4_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 600, by = 200), expand = c(0, 0), limits = c(0, 600)) +
  ggtitle("DHL4 PS Around Func Motif") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
DHL4_hist

NUD_hist_peak <- ggplot (all, aes(x = bin)) +
  geom_line(aes(y = NUD_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = NUD_0.25_A, color = "15 min"), size = 0.75) +
  geom_line(aes(y = NUD_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = NUD_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = NUD_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 500, by = 250), expand = c(0, 0), limits = c(0, 500)) +
  ggtitle("NUDUL1 PS Around Func Peaks") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
NUD_hist_peak

LY1_hist_peak <- ggplot (all_peak, aes(x = bin)) +
  geom_line(aes(y = LY1_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = LY1_0.25_A, color = "15 min"), size = 0.75) +
  geom_line(aes(y = LY1_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = LY1_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = LY1_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 500, by = 250), expand = c(0, 0), limits = c(0, 500)) +
  ggtitle("LY1 PS Around Func Peaks") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
LY1_hist_peak

DHL4_hist_peak <- ggplot (all, aes(x = bin)) +
  geom_line(aes(y = DHL4_0_A, color = "0 hr"), size = 0.75) +
  geom_line(aes(y = DHL4_0.5_A, color = "30 min"), size = 0.75) +
  geom_line(aes(y = DHL4_1_A, color = "1 hr"), size = 0.75) +
  geom_line(aes(y = DHL4_2_A, color = "2 hr"), size = 0.75) +
  scale_x_continuous(breaks = seq(1, 601, by = 300), expand = c(0, 0), limits = c(1, 601)) +
  scale_y_continuous(breaks = seq(0, 500, by = 250), expand = c(0, 0), limits = c(0, 500)) +
  ggtitle("DHL4 PS Around Func PEAKs") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
DHL4_hist_peak
```

## Save to png
```{r, fig.show= "hide"}
png <- function(name, data) {
ggplot2::ggsave(file= name, 
             device="png",
             scale = 1,
             dpi=300,
             width = 5,
             height = 4,
             plot = data)
}

smallpng <- function(name, data) {
ggplot2::ggsave(file= name, 
             device="png",
             scale = 1,
             dpi=300,
             width = 3.5,
             height = 2.5,
             plot = data)
}

png("20230826_histo_ATACseqsig_NUD_aorundfuncFKHmotifs_bin25_3kb_5x4.png", NUD_hist)
png("20230826_histo_ATACseqsig_LY1_aorundfuncFKHmotifs_bin25_3kb_5x4.png", LY1_hist)
png("20230826_histo_ATACseqsig_DHL4_aorundfuncFKHmotifs_bin25_3kb_5x4.png", DHL4_hist)

smallpng("20230826_histo_ATACseqsig_NUD_aorundfuncFKHmotifs_bin25_3kb_3.5x2.5.png", NUD_hist)
smallpng("20230826_histo_ATACseqsig_LY1_aorundfuncFKHmotifs_bin25_3kb_3.5x2.5.png", LY1_hist)
smallpng("20230826_histo_ATACseqsig_DHL4_aorundfuncFKHmotifs_bin25_3kb_3.5x2.5.png", DHL4_hist)

png("20230826_histo_ATACseqsig_NUD_aorundfuncFKHpeaks_bin25_3kb_5x4.png", NUD_hist_peak)
png("20230826_histo_ATACseqsig_LY1_aorundfuncFKHpeaks_bin25_3kb_5x4.png", LY1_hist_peak)
png("20230826_histo_ATACseqsig_DHL4_aorundfuncFKHpeaks_bin25_3kb_5x4.png", DHL4_hist_peak)

smallpng("20230826_histo_ATACseqsig_NUD_aorundfuncFKHpeaks_bin25_3kb_3.5x2.5.png", NUD_hist_peak)
smallpng("20230826_histo_ATACseqsig_LY1_aorundfuncFKHpeaks_bin25_3kb_3.5x2.5.png", LY1_hist_peak)
smallpng("20230826_histo_ATACseqsig_DHL4_aorundfuncFKHpeaks_bin25_3kb_3.5x2.5.png", DHL4_hist_peak)
```

## PRO-seq
Histograms for PRO-seq data were created in bash and R.
### Calculate signal around peaks
The signal around the desired peak set was calculated using HOMER in bash
```
annotatePeaks.pl ../../Hillary_X6/Nud_DHL4_ATAC_forTobias/TF_footprinting_out/20230801_funcFKHmotifs_motifonly.bed hg19 -size 6000 -hist 10 -d TagDir_0hr/ TagDir_1hr/ TagDir_2hr/ ../../Hillary-2/7771-LY1_FOXO1_PROseq_newprotocol_comb/LY1-FOXO1_0hr_comb/ ../../Hillary-2/7771-LY1_FOXO1_PROseq_newprotocol_comb/LY1-FOXO1_0.5hr_comb/ ../../Hillary-2/7771-LY1_FOXO1_PROseq_newprotocol_comb/LY1-FOXO1_1hr_comb/ ../../Hillary-2/7771-LY1_FOXO1_PROseq_newprotocol_comb/LY1-FOXO1_2hr_comb/ ../../Clare/Hillary/3466_DHL4_FOXO1_PROseq_2/TagDir_0hr_combined/ ../../Clare/Hillary/3466_DHL4_FOXO1_PROseq_2/TagDir_0.5hr_combined/ ../../Clare/Hillary/3466_DHL4_FOXO1_PROseq_2/TagDir_1hr_combined/ ../../Clare/Hillary/3466_DHL4_FOXO1_PROseq_2/TagDir_2hr_combined/ > 20230824_hist_PS_aroundfuncFKHmotifs_NUD_LY1_DHL4.txt
```
### Plot histograms
Final histograms were plotted in R.


#### Load libraries
```{r}
library(tidyverse)
library(hillaryscolors)
library(viridis)
```

#### Load Files
```{r}
NUD <- read.delim("20230703_NUDUL1_PSaroundfuncfkh_hist25.txt")
LY1 <- read.delim("20230703_LY1_PSaroundfuncfkh_hist25.txt")
DHL4 <- read.delim("20230703_DHL4_PSaroundfuncfkh_hist25.txt")
```

#### Make histograms
```{r}
## Rearrange data
NUD_adj <- NUD %>%
  dplyr::select(Distance = 1, T0 = 2, T1 = 5, T2 = 8) 

LY1_adj <- LY1 %>%
  dplyr::select(Distance = 1, T0 = 2, T0.5 = 5, T1 = 8, T2 = 11) 

DHL4_adj <- DHL4 %>%
  dplyr::select(Distance = 1, T0 = 2, T0.5 = 5, T1 = 8, T2 = 11) 

## load histpalette
viridis::viridis(5)

histpalette <- c("0 hr" = "#440154FF", "30 min" = "#3B528BFF", "1 hr" = "#21908CFF", "2 hr" = "#5DC863FF")

## Make histos
NUD_hist <- ggplot (NUD_adj, aes(x = Distance)) +
  geom_line(aes(y = T0, color = "0 hr"), size = 1) +
  geom_line(aes(y = T1, color = "1 hr"), size = 1) +
  geom_line(aes(y = T2, color = "2 hr"), size = 1) +
  scale_x_continuous(breaks = seq(-3000, 3000, by = 3000), expand = c(0, 0), limits = c(-3000, 3000)) +
  scale_y_continuous(breaks = seq(0, 6, by = 2), expand = c(0, 0), limits = c(0, 6)) +
  ggtitle("NUDUL1 PS Around Func FKH") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
NUD_hist

LY1_hist <- ggplot (LY1_adj, aes(x = Distance)) +
  geom_line(aes(y = T0, color = "0 hr"), size = 1) +
  geom_line(aes(y = T0.5, color = "30 min"), size = 1) +
  geom_line(aes(y = T1, color = "1 hr"), size = 1) +
  geom_line(aes(y = T2, color = "2 hr"), size = 1) +
  scale_x_continuous(breaks = seq(-3000, 3000, by = 3000), expand = c(0, 0), limits = c(-3000, 3000)) +
  scale_y_continuous(breaks = seq(0, 6, by = 2), expand = c(0, 0), limits = c(0, 6)) +
  ggtitle("LY1 PS Around Func FKH") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
LY1_hist

DHL4_hist <- ggplot (DHL4_adj, aes(x = Distance)) +
  geom_line(aes(y = T0, color = "0 hr"), size = 1) +
  geom_line(aes(y = T0.5, color = "30 min"), size = 1) +
  geom_line(aes(y = T1, color = "1 hr"), size = 1) +
  geom_line(aes(y = T2, color = "2 hr"), size = 1) +
  scale_x_continuous(breaks = seq(-3000, 3000, by = 3000), expand = c(0, 0), limits = c(-3000, 3000)) +
  scale_y_continuous(breaks = seq(0, 10, by = 5), expand = c(0, 0), limits = c(0, 10)) +
  ggtitle("DHL4 PS Around Func FKH") +
  labs(x = "Distance from Peak Center", y = "Fragment Depth (per bp, per peak)") +
  theme_hillary() +
  coord_cartesian(clip = "off") +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold")) +
  scale_color_manual(values = histpalette)
DHL4_hist
```

#### Save to png
```{r, fig.show= "hide"}
png <- function(name, data) {
Cairo::Cairo(file = name, 
             bg = "white",
             type = "png",
             units = "in", 
             width = 8, 
             height = 6, 
             pointsize = 12, 
             dpi = 300)
data
}

png("20230704_histo_PROseqsig_NUD_aorundfuncFKHpeaks_bin25_3kb_8x6.png", NUD_hist)
dev.off()
png("20230704_histo_PROseqsig_LY1_aorundfuncFKHpeaks_bin25_3kb_8x6.png", LY1_hist)
dev.off()
png("20230704_histo_PROseqsig_DHL4_aorundfuncFKHpeaks_bin25_3kb_8x6.png", DHL4_hist)
dev.off()
```

