# 6C. Time course volcano plots
The following code is an example of the code used to generate volcano plots from changes in TF binding from ATAC-seq data in bash and R.
## BINDetect
Changes in transcription factor binding were calculated using TOBIAS in bash.
```
PEAK_DIR=/Volumes/Hillary_X6/Nud_DHL4_ATAC_forTobias/
PROCESSED_DIR=/Volumes/Hillary_X6/Nud_DHL4_ATAC_forTobias/TF_footprinting_out/Functional_peaks/
GENOME_DIR=/Volumes/Hillary_X6/genomes/

ulimit -n 4056

for i in 0.25h 0.5h 1h 2h
do

TOBIAS BINDetect --motifs  ${PEAK_DIR}/coremotifs_jaspar_plusDIV2.txt --signals ${PEAK_DIR}/TF_footprinting_out/NUDUL1_ATAC_0h_footprints.bw ${PEAK_DIR}/TF_footprinting_out/NUDUL1_ATAC_${i}_footprints.bw --genome ${GENOME_DIR}/hg19/hg19.fa --peaks ${PEAK_DIR}/TF_footprinting_out/20230703_funcFKHmotifs_nohead.bed --outdir ${PROCESSED_DIR}/motifs_NUDUL1_${i}vs0h --cond_names 0hr ${i} --cores 8 --verbosity 1
done
```
## Volcano plots
Volcano plots were created in R.

### Load Libraries
```{r}
library(tidyverse)
library(hillaryscolors)
library(stringi)
library(ggrepel)
library(ComplexHeatmap)
library(circlize)
```

### Load bindetect results from TOBIAS and expressed genes
```{r}
bindetect_NUDUL1_0v0.25 <- readr::read_delim("bindetect_results_NUDUL1_0.25v0h.txt")
bindetect_NUDUL1_0v0.5 <- readr::read_delim("bindetect_results_NUDUL1_0.5v0h.txt")
bindetect_NUDUL1_0v1 <- readr::read_delim("bindetect_results_NUDUL1_1v0h.txt")
bindetect_NUDUL1_0v2 <- readr::read_delim("bindetect_results_NUDUL1_2v0h.txt")

bindetect_LY1_0v0.25 <- readr::read_delim("bindetect_results_LY1_0.25v0h.txt")
bindetect_LY1_0v0.5 <- readr::read_delim("bindetect_results_LY1_0.5v0h.txt")
bindetect_LY1_0v1 <- readr::read_delim("bindetect_results_LY1_1v0h.txt")
bindetect_LY1_0v2 <- readr::read_delim("bindetect_results_LY1_2v0h.txt")

bindetect_DHL4_0v0.5 <- readr::read_delim("bindetect_results_DHL4_0.5v0h.txt")
bindetect_DHL4_0v1 <- readr::read_delim("bindetect_results_DHL4_1v0h.txt")
bindetect_DHL4_0v2 <- readr::read_delim("bindetect_results_DHL4_2v0h.txt")

expressed_genes_NUDUL1 <- readr::read_delim("../../Exp 0460 NUDUL1_FOXO1_PS/20230205 NUDUL1_FOXO1_PS_expressedgenes_duprm_manfilter.txt")
expressed_genes_LY1 <- readr::read_delim("../../Exp 0358  PS_LY1_FOXO1_JE/Comb_seqruns/20220912 LY1_FOXO1_PS_expressedgenes_duprm_manfilter.txt")
expressed_genes_DHL4 <- readr::read_delim("../../Exp 0071/20230530 DHL4_FOXO1_PS_expressedgenes_duprm_manfilter.txt")

TF_families <- readr::read_delim("../TF_footprinting/TF_families.txt") %>%
  select(Name, DBD) %>%
  add_row(tibble(Name = "DIV2", DBD = "Forkhead"))
```

### Select genes expressed in all three cell lines
```{r}
expressed_all <- expressed_genes_LY1 %>%
  inner_join(expressed_genes_NUDUL1, by = "Transcript") %>%
  inner_join(expressed_genes_DHL4, by = "Transcript")
```

### Adjust file (L2FC *-1 so that values are  1hr up/ 1 hr down instead of 0hr up /1 hr up) add FOX annotations
```{r}
tf_results_adj <- function(df, changecol, pcol) {
  df %>%
  dplyr::mutate(change = changecol * -1) %>%
  dplyr::mutate(pval = pcol) %>%
  dplyr::mutate(motif = str_extract(name, "((?<=\\.[:digit:].)[[:print:]]+)")) %>%
  dplyr::mutate(motif = replace_na(motif, "DIV2")) %>%
  dplyr::filter(motif %in% expressed_all$Gene | motif == "DIV2") %>%
  dplyr::mutate(forkhead = case_when(str_detect(motif, "FOX") ~ "FKH", str_detect(motif, "DIV2") ~ "DIV2", str_detect(motif, "FOX", negate = TRUE) ~ "NA")) %>%
  dplyr::filter(motif_id != "MA0032.1" & motif_id != "MA0033.1" & motif_id != "MA0479.1")   %>%
  dplyr::mutate(top2.5percent_pval = quantile(pval, probs = 0.025)) %>%
  dplyr::mutate(top2.5percent_l2fc = quantile(change, probs = 0.025)) %>%
  dplyr::mutate(top5percent_pval = quantile(pval, probs = 0.05)) %>%
  dplyr::mutate(top5percent_l2fc = quantile(change, probs = 0.05)) %>%
  dplyr::mutate(bottom2.5percent_l2fc = quantile(change, probs = 0.975)) %>%
  dplyr::mutate(bottom5percent_l2fc = quantile(change, probs = 0.95)) %>%
  dplyr::mutate(sig_2.5 = case_when(pval <= top2.5percent_pval & change > 0 ~ "Up", pval <= top2.5percent_pval &  change < 0 ~ "Down", pval < 1 & change <= top2.5percent_l2fc ~ "Down", pval < 1 &  change >= bottom2.5percent_l2fc ~ "Up", TRUE ~ "zNS")) %>%
  dplyr::mutate(sig_5 = case_when(pval <= top5percent_pval & change > 0 ~ "Up", pval <= top5percent_pval &  change < 0 ~ "Down", pval < 1 & change <= top5percent_l2fc ~ "Down", pval < 1 &  change >= bottom5percent_l2fc ~ "Up", TRUE ~ "zNS"))

}

LY1_0.25h <- tf_results_adj(bindetect_LY1_0v0.25, bindetect_LY1_0v0.25$`0hr_15min_change`, bindetect_LY1_0v0.25$`0hr_15min_pvalue`)
LY1_0.5h <- tf_results_adj(bindetect_LY1_0v0.5, bindetect_LY1_0v0.5$`0hr_30min_change`, bindetect_LY1_0v0.5$`0hr_30min_pvalue`)
LY1_1h <- tf_results_adj(bindetect_LY1_0v1, bindetect_LY1_0v1$`0hr_1hr_change`, bindetect_LY1_0v1$`0hr_1hr_pvalue`)
LY1_2h <- tf_results_adj(bindetect_LY1_0v2, bindetect_LY1_0v2$`0hr_2hr_change`, bindetect_LY1_0v2$`0hr_2hr_pvalue`)

NUDUL1_0.25h <- tf_results_adj(bindetect_NUDUL1_0v0.25, bindetect_NUDUL1_0v0.25$`0hr_0.25h_change`, bindetect_NUDUL1_0v0.25$`0hr_0.25h_pvalue`)
NUDUL1_0.5h <- tf_results_adj(bindetect_NUDUL1_0v0.5, bindetect_NUDUL1_0v0.5$`0hr_0.5h_change`,  bindetect_NUDUL1_0v0.5$`0hr_0.5h_pvalue`)
NUDUL1_1h <- tf_results_adj(bindetect_NUDUL1_0v1, bindetect_NUDUL1_0v1$`0hr_1h_change`,  bindetect_NUDUL1_0v1$`0hr_1h_pvalue`)
NUDUL1_2h <- tf_results_adj(bindetect_NUDUL1_0v2, bindetect_NUDUL1_0v2$`0hr_2h_change`,  bindetect_NUDUL1_0v2$`0hr_2h_pvalue`)

DHL4_0.5h <- tf_results_adj(bindetect_DHL4_0v0.5, bindetect_DHL4_0v0.5$`0hr_0.5h_change`,  bindetect_DHL4_0v0.5$`0hr_0.5h_pvalue`)
DHL4_1h <- tf_results_adj(bindetect_DHL4_0v1, bindetect_DHL4_0v1$`0hr_1h_change`,  bindetect_DHL4_0v1$`0hr_1h_pvalue`)
DHL4_2h <- tf_results_adj(bindetect_DHL4_0v2, bindetect_DHL4_0v2$`0hr_2h_change`,  bindetect_DHL4_0v2$`0hr_2h_pvalue`)
```

### Make volcano plot
```{r}
volplot <- function(df, title, ylimhi) {
  ggplot(df %>% dplyr::arrange(desc(sig_2.5))) +
  geom_point(aes(x = change, y = -log10(pval), fill = sig_2.5, colour = sig_2.5), shape = 21,  size = 4) +
  geom_text_repel(data = subset(df, sig_2.5 !="zNS"), aes(x = change, y = -log10(pval), label = motif), size = 4, box.padding = unit(0.3, "lines"), point.padding = unit(0.3, "lines"), segment.size = 0.25, max.overlaps = Inf, min.segment.length = 0, point.size = 4)+
  ggtitle(title) +
  xlab("Change in Binding Score") +
  ylab("-log10(p-value)") +
  scale_y_continuous(limits = c(0, ylimhi), expand = c(0,0)) +
  scale_x_continuous(limits = c(-1, 1)) +
  theme_hillary()+
  coord_cartesian(clip = "off") +
  scale_fill_manual(values = c("#31607f", "#70000C", "#B3B6B7")) +
  scale_color_manual(values = c("#31607f", "#70000C", "#B3B6B7")) +
  theme(plot.title = element_text(hjust = 0.5, size = "20", face = "bold")) +
  theme(plot.title = element_text(vjust = 1, size = "20", face = "bold"))
}

LY1_0.25h_vol <- volplot(LY1_0.25h, "LY1 0.25 hour functional peaks", 150)
LY1_0.25h_vol
LY1_0.5h_vol <- volplot(LY1_0.5h, "LY1 0.5 hour functional peaks", 150)
LY1_0.5h_vol
LY1_1h_vol <- volplot(LY1_1h, "LY1 1 hour functional peaks", 150)
LY1_1h_vol
LY1_2h_vol <- volplot(LY1_2h, "LY1 2 hour functional peaks", 150)
LY1_2h_vol

NUDUL1_0.25h_vol <- volplot(NUDUL1_0.25h, "NUDUL1 0.25 hour functional peaks", 150)
NUDUL1_0.25h_vol
NUDUL1_0.5h_vol <- volplot(NUDUL1_0.5h, "NUDUL1 0.5 hour functional peaks", 150)
NUDUL1_0.5h_vol
NUDUL1_1h_vol <- volplot(NUDUL1_1h, "NUDUL1 1 hour functional peaks", 150)
NUDUL1_1h_vol
NUDUL1_2h_vol <- volplot(NUDUL1_2h, "NUDUL1 2 hour functional peaks", 150)
NUDUL1_2h_vol

DHL4_0.5h_vol <- volplot(DHL4_0.5h, "DHL4 0.5 hour functional peaks", 150)
DHL4_0.5h_vol
DHL4_1h_vol <- volplot(DHL4_1h, "DHL4 1 hour functional peaks", 150)
DHL4_1h_vol
DHL4_2h_vol <- volplot(DHL4_2h, "DHL4 2 hour functional peaks", 150)
DHL4_2h_vol
```

### Write to PNG
```{r}
png_6x6 <- function(title, plot) {
  Cairo::Cairo(file=title, 
             bg="white",
             type="png",
             units="in", 
             width=7, 
             height=6, 
             pointsize=12, 
             dpi=300)
plot
}

png_6x6("20230731_LY1_0.25h_TF_footprinting_functional_peaks_volplot.png", LY1_0.25h_vol)
dev.off()
png_6x6("20230731_LY1_0.5h_TF_footprinting_functional_peaks_volplot.png", LY1_0.5h_vol)
dev.off()
png_6x6("20230731_LY1_1h_TF_footprinting_functional_peaks_volplot.png", LY1_1h_vol)
dev.off()
png_6x6("20230731_LY1_2h_TF_footprinting_functional_peaks_volplot.png", LY1_2h_vol)
dev.off()

png_6x6("20230731_NUDUL1_0.25h_TF_footprinting_functional_peaks_volplot.png", NUDUL1_0.25h_vol)
dev.off()
png_6x6("20230731_NUDUL1_0.5h_TF_footprinting_functional_peaks_volplot.png", NUDUL1_0.5h_vol)
dev.off()
png_6x6("20230731_NUDUL1_1h_TF_footprinting_functional_peaks_volplot.png", NUDUL1_1h_vol)
dev.off()
png_6x6("20230731_NUDUL1_2h_TF_footprinting_functional_peaks_volplot.png", NUDUL1_2h_vol)
dev.off()

png_6x6("20230731_DHL4_0.5h_TF_footprinting_functional_peaks_volplot.png", DHL4_0.5h_vol)
dev.off()
png_6x6("20230731_DHL4_1h_TF_footprinting_functional_peaks_volplot.png", DHL4_1h_vol)
dev.off()
png_6x6("20230731_DHL4_2h_TF_footprinting_functional_peaks_volplot.png", DHL4_2h_vol)
dev.off()


svg_30x25 <- function(title, plot) {
  Cairo::Cairo(file=title, 
             bg="white",
             type="svg",
             units="in", 
             width=30, 
             height=25, 
             pointsize=12, 
             dpi=300)
plot
}

svg_30x25("20230731_LY1_0.25h_TF_footprinting_functional_peaks_volplot.svg", LY1_0.25h_vol)
dev.off()
svg_30x25("20230731_LY1_0.5h_TF_footprinting_functional_peaks_volplot.svg", LY1_0.5h_vol)
dev.off()
svg_30x25("20230731_LY1_1h_TF_footprinting_functional_peaks_volplot.svg", LY1_1h_vol)
dev.off()
svg_30x25("20230731_LY1_2h_TF_footprinting_functional_peaks_volplot.svg", LY1_2h_vol)
dev.off()

svg_30x25("20230731_NUDUL1_0.25h_TF_footprinting_functional_peaks_volplot.svg", NUDUL1_0.25h_vol)
dev.off()
svg_30x25("20230731_NUDUL1_0.5h_TF_footprinting_functional_peaks_volplot.svg", NUDUL1_0.5h_vol)
dev.off()
svg_30x25("20230731_NUDUL1_1h_TF_footprinting_functional_peaks_volplot.svg", NUDUL1_1h_vol)
dev.off()
svg_30x25("20230731_NUDUL1_2h_TF_footprinting_functional_peaks_volplot.svg", NUDUL1_2h_vol)
dev.off()

svg_30x25("20230731_DHL4_0.5h_TF_footprinting_functional_peaks_volplot.svg", DHL4_0.5h_vol)
dev.off()
svg_30x25("20230731_DHL4_1h_TF_footprinting_functional_peaks_volplot.svg", DHL4_1h_vol)
dev.off()
svg_30x25("20230731_DHL4_2h_TF_footprinting_functional_peaks_volplot.svg", DHL4_2h_vol)
dev.off()
```
