# 5F. Heatmap of change in TFBS
The following code was used to create a heatmap of the chage in TFBS over time in R.
## Load Libraries
```{r}
library(tidyverse)
library(hillaryscolors)
library(stringi)
library(ComplexHeatmap)
library(circlize)
library(paletteer)
```

## Load annotated peaks file (from Homer) and extract simplified annotations to column anno
```{r}
bindetect_NUDUL1_0v0.25 <- readr::read_delim("bindetect_results_NUDUL1_0v0.25.txt")
bindetect_NUDUL1_0v0.5 <- readr::read_delim("bindetect_results_NUDUL1_0.5.txt")
bindetect_NUDUL1_0v1 <- readr::read_delim("bindetect_results_NUDUL1_0v1.txt")
bindetect_NUDUL1_0v2 <- readr::read_delim("bindetect_results_NUDUL1_0v2.txt")

bindetect_LY1_0v0.25 <- readr::read_delim("bindetect_results_LY1_0v0.25.txt")
bindetect_LY1_0v0.5 <- readr::read_delim("bindetect_results_LY1_0v0.5.txt")
bindetect_LY1_0v1 <- readr::read_delim("bindetect_results_LY1_0v1.txt")
bindetect_LY1_0v2 <- readr::read_delim("bindetect_results_LY1_0v2.txt")

bindetect_DHL4_0v0.5 <- readr::read_delim("bindetect_results_DHL4_0v0.5.txt")
bindetect_DHL4_0v1 <- readr::read_delim("bindetect_results_DHL4_0v1.txt")
bindetect_DHL4_0v2 <- readr::read_delim("bindetect_results_DHL4_0v2.txt")

expressed_genes_NUDUL1 <- readr::read_delim("../../Exp 0460 NUDUL1_FOXO1_PS/20230205 NUDUL1_FOXO1_PS_expressedgenes_duprm_manfilter.txt")
expressed_genes_LY1 <- readr::read_delim("../../Exp 0358  PS_LY1_FOXO1_JE/Comb_seqruns/20220912 LY1_FOXO1_PS_expressedgenes_duprm_manfilter.txt")
expressed_genes_DHL4 <- readr::read_delim("../../Exp 0071/20230530 DHL4_FOXO1_PS_expressedgenes_duprm_manfilter.txt")

TF_families <- readr::read_delim("TF_families.txt") %>%
  select(Name, DBD) %>%
  add_row(tibble(Name = "DIV2", DBD = "Forkhead"))
```
## Select genes expressed in all three cell lines
```{r}
expressed_all <- expressed_genes_LY1 %>%
  inner_join(expressed_genes_NUDUL1, by = "Transcript") %>%
  inner_join(expressed_genes_DHL4, by = "Transcript")
```

## Adjust file (L2FC *-1 so that values are  1hr up/ 1 hr down instead of 0hr up /1 hr up) add FOX annotations
```{r}
tf_results_adj <- function(df, changecol, pcol) {
  df %>%
  dplyr::mutate(change = changecol * -1) %>%
  dplyr::mutate(pval = pcol) %>%
  dplyr::mutate(motif = str_extract(name, "((?<=\\.[:digit:].)[[:print:]]+)")) %>%
  dplyr::mutate(motif = replace_na(motif, "DIV2")) %>%
  dplyr::filter(motif %in% expressed_all$Gene | motif == "DIV2") %>%
  dplyr::left_join(TF_families, by = c("motif" = "Name")) %>%
  dplyr::mutate(forkhead = case_when(str_detect(motif, "FOX") ~ "FKH", str_detect(motif, "DIV2") ~ "DIV2", str_detect(motif, "FOX", negate = TRUE) ~ "NA")) %>%
  dplyr::filter(motif_id != "MA0032.1" & motif_id != "MA0033.1" & motif_id != "MA0479.1")
}

LY1_0.25h <- tf_results_adj(bindetect_LY1_0v0.25, bindetect_LY1_0v0.25$`0hr_0.25hr_change`, bindetect_LY1_0v0.25$`0hr_0.25hr_pvalue`)
LY1_0.5h <- tf_results_adj(bindetect_LY1_0v0.5, bindetect_LY1_0v0.5$`0hr_0.5hr_change`, bindetect_LY1_0v0.5$`0hr_0.5hr_pvalue`)
LY1_1h <- tf_results_adj(bindetect_LY1_0v1, bindetect_LY1_0v1$`0hr_1hr_change`, bindetect_LY1_0v1$`0hr_1hr_pvalue`)
LY1_2h <- tf_results_adj(bindetect_LY1_0v2, bindetect_LY1_0v2$`0hr_2hr_change`, bindetect_LY1_0v2$`0hr_2hr_pvalue`)

NUDUL1_0.25h <- tf_results_adj(bindetect_NUDUL1_0v0.25, bindetect_NUDUL1_0v0.25$`0hr_0.25h_change`, bindetect_NUDUL1_0v0.25$`0hr_0.25h_pvalue`)
NUDUL1_0.5h <- tf_results_adj(bindetect_NUDUL1_0v0.5, bindetect_NUDUL1_0v0.5$`0hr_0.5h_change`,  bindetect_NUDUL1_0v0.5$`0hr_0.5h_pvalue`)
NUDUL1_1h <- tf_results_adj(bindetect_NUDUL1_0v1, bindetect_NUDUL1_0v1$`0hr_1h_change`,  bindetect_NUDUL1_0v1$`0hr_1h_pvalue`)
NUDUL1_2h <- tf_results_adj(bindetect_NUDUL1_0v2, bindetect_NUDUL1_0v2$`0hr_2h_change`,  bindetect_NUDUL1_0v2$`0hr_2h_pvalue`)

DHL4_0.5h <- tf_results_adj(bindetect_DHL4_0v0.5, bindetect_DHL4_0v0.5$`0hr_0.5h_change`,  bindetect_DHL4_0v0.5$`0hr_0.5h_pvalue`)
DHL4_1h <- tf_results_adj(bindetect_DHL4_0v1, bindetect_DHL4_0v1$`0hr_1h_change`,  bindetect_DHL4_0v1$`0hr_1h_pvalue`)
DHL4_2h <- tf_results_adj(bindetect_DHL4_0v2, bindetect_DHL4_0v2$`0hr_2h_change`,  bindetect_DHL4_0v2$`0hr_2h_pvalue`)
```
## Make Heatmap
```{r}
heatmap_df <- NUDUL1_0.25h %>%
  left_join(NUDUL1_0.5h, by = "motif_id") %>%
  left_join(NUDUL1_1h, by = "motif_id") %>%
  left_join(NUDUL1_2h, by = "motif_id") %>%
  left_join(LY1_0.25h, by = "motif_id") %>%
  left_join(LY1_0.5h, by = "motif_id") %>%
  left_join(LY1_1h, by = "motif_id") %>%
  left_join(LY1_2h, by = "motif_id") %>%
  left_join(DHL4_0.5h, by = "motif_id") %>%
  left_join(DHL4_1h, by = "motif_id") %>%
  left_join(DHL4_2h, by = "motif_id") %>%
  select(motif_id, motif.x, DBD.x, NUD_0.25 = change.x, NUD_0.5 = change.y, NUD_1 = change.x.x, NUD_2 = change.y.y, LY1_0.25 = change.x.x.x, LY1_0.5 = change.y.y.y, LY1_1 = change.x.x.x.x, LY1_2 = change.y.y.y.y, DHL4_0.5 = change.x.x.x.x.x, DHL4_1 = change.y.y.y.y.y, DHL4_2 = change) %>%
  arrange(DBD.x, motif.x, .locale = "en")

heatmap_df <-as.data.frame(heatmap_df)

rnames = heatmap_df [ , 2]
mat_data = data.matrix(heatmap_df [ , 4:14])
rownames(mat_data) = rnames

my_palette = colorRamp2(c(-0.5, 0, 0.5), c("#053061", "#f5f5f5", "#70000C"))

DBD_mat = data.matrix(heatmap_df [ , 3])

heatmap_main <- Heatmap(mat_data, col = my_palette, column_title = "Changein TF Binding", cluster_columns = FALSE, cluster_rows = FALSE, show_row_names = TRUE, show_column_names = TRUE, row_names_gp = gpar(fontsize = 4), border = TRUE) 
heatmap_main

colors <- c(paletteer_d("ggthemes::manyeys", 19), paletteer_d("ggthemes::manyeys", 13)) 

side_pal = structure(colors, names = c("AP-2", "bHLH", "bZIP", "C2H2 ZF", "C2H2 ZF; AT hook", "C2H2 ZF; Homeodomain", "CSL", "CUT; Homeodomain", "E2F", "EBF1", "Ets", "Forkhead",  "Grainyhead", "HMG/Sox", "Homeodomain", "Homeodomain; POU", "HSF", "IRF", "MADS box", "Myb/SANT", "Nuclear receptor", "p53", "Paired box", "Rel",  "RFX", "Runt", "SAND", "SMAD", "STAT", "T-box", "THAP finger", "Unknown"))

heatmap_side <- Heatmap(DBD_mat, col = side_pal, column_title = "DBD", cluster_columns = FALSE, show_column_names = FALSE, cluster_rows = FALSE, show_row_names = TRUE, row_names_gp = gpar(fontsize = 10), width = unit(2.5, "mm"), border = TRUE) 
heatmap_side

annotated_dbd_heatmap = heatmap_side + heatmap_main
annotated_dbd_heatmap_draw <- draw(annotated_dbd_heatmap, ht_gap = unit(0.5, "mm"))

annotated_dbd_heatmap_main_side = heatmap_main + heatmap_side
annotated_dbd_heatmap_main_side_draw <- draw(annotated_dbd_heatmap_main_side, auto_adjust = FALSE, ht_gap = unit(0.5, "mm"))

Cairo::Cairo(file = "20230618 ATAC_TF_footprinting_consensu_peaks_allexpressed_TFs_bindingscore_10x10.png", 
             bg = "white",
             type = "png",
             units = "in", 
             width = 10, 
             height = 10, 
             pointsize = 12, 
             dpi = 300)
annotated_dbd_heatmap_main_side_draw
dev.off()
```
